tests/app/test_jinja_filters.py

```diff
@@ -460,3 +375,106 @@ def test_format_address_fields_with_uprn():
     assert (
         get_formatted_address(address_fields) == "7 Evelyn Street<br>Barry<br>CF63 4JG"
     )
+
+
+def _get_test_field_from_answer_schema(answer_schema):
+    questionnaire_schema = Mock()
+    questionnaire_schema.error_messages = error_messages
+    question_schema = {"answers": [answer_schema]}
+    form = generate_form(
+        questionnaire_schema,
+        question_schema,
+        answer_store=Mock(),
+        metadata={},
+        location=None,
+        data=None,
+        form_data=None,
+    )
+
+    return getattr(form, answer_schema["id"])
```

Comment:
We shouldn't be introducing dependencies on some quite involved classes. In the previous tests the form was mocked.

Reply:
Ya its because previous the form wasn't being used in the test, i.e only visibility was tested. Maybe I should only test visibility and input type etc giving these are exisiting.